controllers/medusa/medusaconfiguration_controller.go

```diff
@@ -74,15 +80,23 @@ func (r *MedusaConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	// Check if the referenced secret exists
 	if configuration.Spec.StorageProperties.StorageSecretRef.Name != "" {
-		err = r.CheckSecretPresence(ctx, r.Client, req, configuration.Spec.StorageProperties.StorageSecretRef.Name)
+		secret, err := r.GetSecret(ctx, r.Client, req, configuration.Spec.StorageProperties.StorageSecretRef.Name)
 		if err != nil {
 			logger.Error(err, "Failed to get MedusaConfiguration referenced secret")
 			configuration.Status.SetCondition(medusav1alpha1.ControlStatusSecretAvailable, metav1.ConditionFalse)
 			configuration.Status.SetConditionMessage(medusav1alpha1.ControlStatusSecretAvailable, err.Error())
 			r.patchStatus(ctx, configuration, patch, logger)
 			return ctrl.Result{}, err
-		} else {
-			configuration.Status.SetCondition(medusav1alpha1.ControlStatusSecretAvailable, metav1.ConditionTrue)
+		}
+		configuration.Status.SetCondition(medusav1alpha1.ControlStatusSecretAvailable, metav1.ConditionTrue)
+
+		patch := client.MergeFrom(secret.DeepCopy())
+		if secret.Labels == nil {
+			secret.Labels = make(map[string]string)
+		}
+		secret.Labels[MedusaStorageSecretIdentifierLabel] = utils.HashNameNamespace(secret.Name, secret.Namespace)
+		if err = r.Client.Patch(ctx, secret, patch); err != nil {
```

Comment:
If the hash already matches the previous one, why would we patch? Also, at least one unit test which guards the implementation?

Reply:
I've made this change, however the unit test I've added is now failing since for some reason the patch gets called, fails, and then isn't called again (despite there being a requeue requested).

I'll have to look at this more tomorrow I think.

Reply:
Ah, I know what it is. Fixing, this will be ready for another review in  a minute.